### ISSUE TITLE
Switching Nvidia Surround OFF doesn't work

### ISSUE THREAD

**Issue by FormelLMS** [terrymacdonald/DisplayMagician#76]
**Describe the bug**
I have made the update to 2.1.5.5 and Switching from Surround to normal mode is not working here

**To Reproduce**
Steps to reproduce the behaviour:
Saved two new Display Profiles. One with Surround and One without. Switching from normal to surround is flawlessly working, but back to normal not. 

**Expected behaviour**
:)

**Screenshots**

**Reporting Logs**
If DisplayMagician starts:
- Open DisplayMagician and perform whatever actions are needed to make the bug occur.
- Click on the Settings button on the main DisplayMagician window
- Click on 'Create a Support Zip File' button
- Save the Support Zip File on your computer.
- Come back here, and upload the Support Zip File through your web browser and attach it to this issue.

**Enviroment (please complete the following information):**
 - Windows Version: 11
 - DisplayMagician Version 2.1.5.5
 - Video Card: Gigabyte 3080ti
 - Number and make/model of monitors: 3xMSI27D
 - Date of last video driver update: 479.29
 - Date of last windows update: 18.12.2021
 - 
[DisplayMagician-Support-20220108-1707.zip](https://github.com/terrymacdonald/DisplayMagician/files/7833530/DisplayMagician-Support-20220108-1707.zip)

**Comment by terrymacdonald** [comment]
@FormelLMS What version of DisplayMagician were you running before you upgraded? That will let me see what differences there are between the versions.

Are you able to please install this test version I just created and see if it has the same problem? If this does still have the same problems then we'll need to troubleshoot further with NVIDIAInfo.
[DisplayMagicianSetup-v2.2.0.4.zip](https://github.com/terrymacdonald/DisplayMagician/files/7834114/DisplayMagicianSetup-v2.2.0.4.zip)

Thanks
Terry

**Comment by terrymacdonald** [comment]
Unfortunately I couldn't see any problems in the DisplayMagician.log file because DisplayMagician starts a brand new log file each time it starts. In versions 2.1.5.5 and earlier you would need to start DisplayMagician change to your "Surround 1440p" profile then back again to your "Normal 1440p" profile, and then go straight to the 'Create Support ZIP file' button and save the zip file in order to save the log file with information about the profile swap in it.

In the test version 2.2.0.4 I gave you, I've now made DisplayMagician automatically keep copies of the last 4 log files, and include them in the Support ZIP file, so you won't need to worry about that from now on at least :). Constant improvement....

So yeah, please swap from your "Surround 1440p" profile then back again to your "Normal 1440p" profile, and then create a support ZIP file for me and upload it again. One of the DisplayMagician.log files in the support ZIP file should give me more information about the issue.

I do think we'll need to do some testing with NVIDIAInfo though based on your description of the issue.  We'll wait until I get to see the DisplayMagician log file before we'll do any NVIDIAInfo testing though.

Thanks
Terry

**Comment by sk666** [comment]
Hi Terry,

if this can help, I have this issue too with 2.2.0.4. I need to quit surround by nvidia control panel. As you can see in this picture, DM start to roll back to 1920x1080 profile but it keep surround instead.

![Capture](https://user-images.githubusercontent.com/29050341/148682884-012f0295-ee87-4b2d-9b99-4cf51378a0b3.PNG)

**Comment by FormelLMS** [comment]
Hey Terry, thanks for helping out. I've tried some things now. Installed the new version. This does the same. Then uninstalled driver via DDU und renewed that all, but that didn't do the trick, too.
Did the log file and dropped it here. It's exactly as sk666 said (Thank You :) )
[DisplayMagician-Support-20220109-1410.zip](https://github.com/terrymacdonald/DisplayMagician/files/7835091/DisplayMagician-Support-20220109-1410.zip)

What I have to say is that driver installing, forcing my 1080p displays to 1440p was a pain in the ass. When switching Surround off in the nvidia driver, the mark remains even after pressing apply. I have to that twice: Uncheck, apply, wait - uncheck, apply. Then the mark is unchecked in driver. 
At second: After rebooting, surround is enabled again. Something seems to be very wrong on the driver and/or Windows 11 here.
@sk666 you using Win11, too?

**Comment by sk666** [comment]
@FormelLMS no I have W11, I use W10. I used DDu myself 2 days ago to fix another issue with DM and this bug happening too so uninstall and reinstall nvidia driver dosen't work for this issue like you said :(

**Comment by FormelLMS** [comment]
> @FormelLMS no I have W10 21H2. I used DDU myself 2 days ago to fix another issue with DM and this bug happening too so uninstall and reinstall nvidia driver dosen't work for this issue like you said :(

Gods news that it seems to be not an Win 11 issue. In my inner eyes I saw me weeks of fiddling with a new windows installation

**Comment by terrymacdonald** [comment]
Hmmmm. I wonder if this is related to the switch to a multithreaded process
too? TBH I've lost track of which version has what, so I'll have to look
into it deeper over the next few days.

I'm currently taking some time to map out the flow of logic and calling
logic for the new multithreaded model we've moved to, and I'll be going
through each statement to figure out where any issues may be. As mentioned
above this could be related to that.

This could take a little while to sort out ☹️.

@FormelLMS, could you please do some testing for me with NVIDIAInfo? I'd
like to see if the NVIDIALibrary I made still works, and NVIDIAInfo allows
us to test that.

I've reused some instructions below. Please run through them exactly as
shown. Note: where it says 'Racing', it means your surround display
layout....

So, can you please download the latest NVIDIAInfo zip from
https://github.com/terrymacdonald/NVIDIAInfo/releases and unzip it to a
test folder. Then open a command line/powershell to that same folder and
run through the following commands. Make sure you do every step in order,
and record the results:

   - Set up your Regular profile setup, and then run NVIDIAInfo save
   regular.cfg. That will save the Regular configuration
   - Save the NVIDIAInfo.log file as something like nv-save-regular.log so
   that you can send it to me later.
   - Test that the Regular configuration was saved correctly by NVIDIAInfo
   equal regular.cfg. Please record if this test was equal or not. We want
   it to be equal.
   - Set up your Racing profile setup (with only the surround), and then
   run NVIDIAInfo save racing.cfg. That will save the Racing configuration
   - Save the NVIDIAInfo.log file as something like nv-save-racing.log so
   that you can send it to me later.
   - Test that the Racing configuration was saved correctly by NVIDIAInfo
   equal racing.cfg. Please record if this test was equal or not. We want
   it to be equal.

At this stage we have the two basic configs set. Next step is to see if it
works changing between them:

   - Change from the Racing profile to the Regular profile by running
NVIDIAInfo
   load regular.cfg. That will try to swap to the Regular configuration.
   - Save the NVIDIAInfo.log file as something like
   nv-change-to-regular1.log so that you can send it to me later.
   - Test that the configuration no longer matches the Racing profile by
   running NVIDIAInfo equal racing.cfg. Please record if this test was
   equal or not. We want it to NOT be equal.
   - Test that the Regular configuration was loaded correctly by NVIDIAInfo
   equal regular.cfg. Please record if this test was equal or not. We want
   it to be equal. This means the profile was matched properly.
   - Change from the Regular profile back to the Racing profile by
running NVIDIAInfo
   load racing.cfg. That will try to swap to the Racing configuration.
   - Save the NVIDIAInfo.log file as something like
   nv-change-back-racing1.log so that you can send it to me later.
   - Test that the configuration no longer matches the Regular profile by
   running NVIDIAInfo equal regular.cfg. Please record if this test was
   equal or not. We want it to NOT be equal.
   - Test that the Racing configuration was loaded correctly by NVIDIAInfo
   equal racing.cfg. Please record if this test was equal or not. We want
   it to be equal. This means the profile was matched properly.

Once you've got to this point, please come back here and tell me the
results, and upload all the log files, and the .cfg files you created. That
will all give me enough information to tell you the next steps.

Thanks
Terry

**Comment by FormelLMS** [comment]
Hey Terry, thanks for looking up here :) 
I did my very best and followed your instructions. I've understood everything you wrote so here are the results:

Saved the regular cfg und it was equal to the actual one. Them switched to my surround mode an made the surround cfg. This was then equal to the profile loaded. But switching back to the regular one doesn't work here, too. I've renamed the logfiles and put everything to the attached zip file.

Hoping, there is something in for you :)
[DisplayMagician Issue76 FormelLMS.zip](https://github.com/terrymacdonald/DisplayMagician/files/7835602/DisplayMagician.Issue76.FormelLMS.zip)

No poblem if it takes a bit longer. Fortunately the switchback to surround is working here. This is the worser part to do manually. Switching back to regular is much easier manually ;)

**Comment by terrymacdonald** [comment]
I *think* I might know what's happening but I'll need to figure it out tonight. It looks from the logs in my testing that the NVIDIALibrary is running in parallel (at the same time) as the WinLibrary functions. This is not supposed to happen - they must run after each other in sequence - so I definitely think it's something that changing to the multithreaded model has caused to happen.

Because each thread is independent, that potentially explains why people are having different problems for each other. Each computer has different hardware that runs at different speeds. This difference in hardware means that each computer gets to different parts of the code at different times, meaning they may clash in different ways. For example in my testing I cannot apply a surround profile properly, but heading back to a non surround profile works fine. @FormelLMS cannot from a surround profile to a non-surround one. 

This may not be the only problem, but it is definitely a contributor to the issues we're seeing. I'll need to map out the thread data flows, and figure out how I'm going to stop this from happening. I need to make the child processes wait for each other.

I'll let you both know when I find the actual root cause.

Thanks
Terry

**Comment by terrymacdonald** [comment]
OK @FormelLMS - I *think* I finally cracked this last night. I had to transition from the original toast notification library I was using to the newer Microsoft one. I think this will work better in the long run, as it will be updated by Microsoft to support their windows Toast improvements as Windows dev continues.

In any case, can you please install this test v2.2.0.41 please? And see if all your errors have disappeared? Please report back with any errors you find (screenshots too please if possible) and of course upload the support ZIP file if you have any errors too.

[DisplayMagicianSetup-v2.2.0.41.zip](https://github.com/terrymacdonald/DisplayMagician/files/7866290/DisplayMagicianSetup-v2.2.0.41.zip)

Thanks
Terry

**Comment by terrymacdonald** [comment]
@FormelLMS I've finally had a chance to look at your NVIDIAInfo logs. There is indeed a problem that NVIDIALibrary complains about: 

```
2022-01-09 18:52:18.7318|WARN|DisplayMagicianShared.SharedLogger|NVIDIALibrary/SetActiveConfig: There was an error disabling the display mode. NvAPI_Mosaic_EnableCurrentTopo() returned error code NVAPI_MODE_CHANGE_FAILED
```

This is the NVIDIA Driver complaining when we try to turn off the NVIDIA Surround (Mosaic). It's saying it can't turn it off! I have NO idea why, as the config that we are currently using has mosaic enabled, and the one we want to use does not have mosaic enabled, so we should be able to turn off the mosaic mode! This should work (it WAS working) -  I can only believe it is a NVIDIA driver error or something. I'll do some more research over the next few days to see if I can add some additional tests in there somehow(?).

Did you install a cut-down version of NVIDIA driver using a custom installer? Or did you install the stock NVIDIA game ready driver from NVIDIA itself?

What version of NVIDIA video card are you running? NVIDIA GTX3070? 3080 Ti? Which manufacturer and model is it? Is it the same one that @sk666 has?

I have just noticed that there is a new NVIDIA Game ready driver 511.23. an you please try and install the stock version of that using the NVIDIA installer and see if that works?

I need one more cfg file from you as well. I need to compare the NVIDIAInfo configuration that we want to apply with the setup that actually results, so can you please run the following commands for me (in the exact order shown)?

Open a powershell terminal window in the same folder as the NVIDIAInfo program, then:
- Run `NVIDIAInfo load surround.cfg` to load the surround cfg
- Rename the NVIDIAInfo.log file to `Rename-Item -Path "NVIDIAInfo.log" -NewName "NVIDIAInfo-surround-load.log"`
- Check if the first windows display config is equal with `NVIDIAInfo equal surround.cfg` (we want it to be equal)
- Please take a screenshot of the config equality test.
- Run `NVIDIAInfo save surround-applied.cfg` to save current surround cfg for me to compare
- Run `NVIDIAInfo load regular.cfg` to load the regular.cfg
- Rename the NVIDIAInfo.log file to `Rename-Item -Path "NVIDIAInfo.log" -NewName "NVIDIAInfo-regular-load.log"`
- Check if the first windows display config is equal with `NVIDIAInfo equal regular.cfg` (we want it to be equal, but it is likely to fail)
- Please take a screenshot of the 2nd config equality test.
- Run `NVIDIAInfo save regular-applied.cfg` to save current regular cfg for me to compare

Next, create a zip archive containing the following files, and upload it here:
- surround.cfg
- surround-applied.cfg
- NVIDIAInfo-surround-load.log
- The surround equality test screenshot
- regular.cfg
- regular-applied.cfg
- NVIDIAInfo-regular-load.log
- The regular equality test screenshot

With that info, I should be able to see what has changed (or not changed) between the different profile changes. It will give me a good idea on what bits I need to look at.

Thanks
Terry

**Comment by terrymacdonald** [comment]
Ah I just looked up the top of the list for the video card. Its a Gigabyte 3080 Ti. No need to include that info in the last response any more :)

**Comment by terrymacdonald** [comment]
@FormelLMS Can you please try those tests I listed above with the new test version of NVIDIAInfo (v1.3.9) which I've linked in this comment? It has an extra step where I try to doublecheck that the mosaic is enabled before I try turning it off. I'm doubtful this will fix anything, but I thought I'd add in the check just in case.

[NVIDIAInfo-v1.3.9.zip](https://github.com/terrymacdonald/DisplayMagician/files/7874631/NVIDIAInfo-v1.3.9.zip)

Everything still works on my system (but it was working before too!). I've never had the issues you've experienced yet, which makes it really hard to troubleshoot.

Thanks
Terry

**Comment by FormelLMS** [comment]
Hey Terry, hope, your doing well. Heard of that volcano and thought of you there. 

I've updated now to the driver you say and the new display magician. This alone doesn't work unfortunately. I always install the nvidia drivers without any exceptions (express install)

So here are the next stepstaken with 1.3.9 of NVIDIAInfo. Hopefully it will help you to try to sort it out :)
Hopefully didn't forgot a file.
[Desktop.zip](https://github.com/terrymacdonald/DisplayMagician/files/7874899/Desktop.zip)

**Comment by terrymacdonald** [comment]
@FormelLMS Thanks for that. It unfortunately shows that the extra layer of protection I added doesn't help. That tells me that The NVIDIA driver knows that Mosaic is enabled, which makes it so more confusing that it won't let us turn Mosaic off :(. 

```
2022-01-15 12:56:54.9091|ERROR|DisplayMagicianShared.SharedLogger|NVIDIALibrary/SetActiveConfig: There was an error disabling the display mode. NvAPI_Mosaic_EnableCurrentTopo() returned error code NVAPI_MODE_CHANGE_FAILED
```

Ok - so time for my next theory. My hypothesis is that *maybe* the Mosaic function has now been updated to doublecheck that the mode it is changing to is a valid mode for windows. If this is the case, then maybe it's best to turn off the Mosaic mode as the first thing we do. At the moment we turn off any NVIDIA custom HDR/Colour settings before we change the Mosaic. NVIDIAInfo v1.3.9b tests that theory, and I've attached it below. We'll see if the that makes the `NvAPI_Mosaic_EnableCurrentTopo(false)` function call work.

[NVIDIAInfo-v1.3.9b.zip](https://github.com/terrymacdonald/DisplayMagician/files/7875909/NVIDIAInfo-v1.3.9b.zip)

Please note that as I've disabled the NVIDIA custom HDR/Colour change code, the equality tests won't work (as the colour will always be wrong). 

Can you please run through these set of tests again please? We'll name these the **'surround off NVIDIAInfo tests'** so that I can just tell you to run the **'surround off NVIDIAInfo tests'** and you'll know what I mean :)

Open a powershell terminal window in the same folder as the NVIDIAInfo program, then:
- Delete the log files from the previous tests: `del NVIDIAInfo-surround-load.log NVIDIAInfo-regular-load.log`
- Run `NVIDIAInfo load surround.cfg` to load the surround cfg
- Rename the NVIDIAInfo.log file to `Rename-Item -Path "NVIDIAInfo.log" -NewName "NVIDIAInfo-surround-load.log"`
- Check if the first windows display config is equal with `NVIDIAInfo equal surround.cfg` (we want it to be equal)
- Please take a screenshot of the config equality test.
- Run `NVIDIAInfo save surround-applied.cfg` to save current surround cfg for me to compare
- Run `NVIDIAInfo load regular.cfg` to load the regular.cfg
- Rename the NVIDIAInfo.log file to `Rename-Item -Path "NVIDIAInfo.log" -NewName "NVIDIAInfo-regular-load.log"`
- Check if the first windows display config is equal with `NVIDIAInfo equal regular.cfg` (we want it to be equal, but it is likely to fail)
- Please take a screenshot of the 2nd config equality test.
- Run `NVIDIAInfo save regular-applied.cfg` to save current regular cfg for me to compare

Next, create a zip archive containing the following files, and upload it here:
- surround.cfg
- surround-applied.cfg
- NVIDIAInfo-surround-load.log
- regular.cfg
- regular-applied.cfg
- NVIDIAInfo-regular-load.log

Let's see if this is the thing causing the issue. I might need to raise a helpdesk ticket with NVIDIA if this doesn't work.

Thanks
Terry

**Comment by FormelLMS** [comment]
Hey Terry,

thanks again for helping here. Unfortunately the problem still exists at the same stop. Loading the regular.cfg doesn't apply here.
I've put the zip file together. Hoing you can find something :)

[surround off NVIDIAInfo tests.zip](https://github.com/terrymacdonald/DisplayMagician/files/7876898/surround.off.NVIDIAInfo.tests.zip)

**Comment by terrymacdonald** [comment]
Hi,

I hope so too! I know where the problem is, but I'm not sure how to work
around it without help from NVIDIA themselves. The difficulty is that the
error occurs on the only 'turn off surround' function they have. I can't
find another way to disable surround, which means we have to try and figure
out why it is happening.

Thanks
Terry

**Comment by FormelLMS** [comment]
Do you think it could be a problem here on my setup in any way? I don’t want to take your time if it’s not on programs side. 
Bit if I can help in any way, I will do 😊

**Comment by terrymacdonald** [comment]
Maybe? There appears to be subset of people with this issue so far, so trying to figure out what the common feature is.

I just had a thought. If I remember correctly your surround set up uses bezel adjustment. Are you able to create a non-bezel adjusted surround  layout and save it as a new NVIDIAInfo cfg? Then try swapping to regular.cfg and see if that works?

Thanks
Terry

**Comment by FormelLMS** [comment]
Terry - unfortunately it's the same.

**Comment by terrymacdonald** [comment]
@FormelLMS Thanks to @erbkaiser I've been doing some other tests in the background, and I've got a hunch that the EnableMosaicTopo error we were getting may be a residual error left over from some of our earlier bugs. So I'd like you to perform the following tests for me:

- Please install DMv2.2.0.52 for me: 
[DisplayMagicianSetup-v2.2.0.52.zip](https://github.com/terrymacdonald/DisplayMagician/files/7887117/DisplayMagicianSetup-v2.2.0.52.zip)
- Please create a brand new Surround Profile and a brand new Triple Screens profile(non-surround).
- Please try to swap between those two new profiles and only those two new profiles (do not use any old profiles at all).
- See if those two new profiles swap without any issue
- Please upload a support ZIP to this issue

I'm hoping that all of the other fixes we have come up with recently will mean that the two newly created profiles will work. I think that the other profiles were created when I still had some multithreading issues, and that may have introduced errors to the display profiles we made then. That of course means that were broken when they were made, which means they will never work (there isn't a way to fix them).

Hopefully, recreating them both again new will mean that they have the correct data in them, and hopefully they will just start working again! Your tests will help say if that's right.

Also - this test version has the 'Send to Clipboard' menu item you wanted in it ;).

Thanks
Terry

**Comment by FormelLMS** [comment]
Hey Terry. At first many thanks for the copy to clipboard. This is working very well. My Backup has included the files already :)

But I have to say that the new version can't change the Surround to normal, too unfortunately. I really has the biggest hopes after your last notification.
[DisplayMagician-Support-20220119-2003.zip](https://github.com/terrymacdonald/DisplayMagician/files/7899942/DisplayMagician-Support-20220119-2003.zip)

**Comment by terrymacdonald** [comment]
Hi @FormelLMS.

Yeah, I figured that would happen, based on @sk666 similar finding. Its
that same error with disabling Mosaic/Surround mode. Looks like we need to
wait for NVIDIA to get back to us.

I've logged a question on their forum, but the person I need to speak with
is apparently on holiday, so it could be a little while.

I also had a look at how Souosh Falahati (original dev of Helios Display
Management) handled disabling surround mode, and he did it the exact same
way that I do it. I was hoping he'd found an alternative way to do it ☹️.

So now we're a bit stuck, and we need to wait for guidance from NVIDIA. My
gut feel is some kind of NVIDIA or windows bug affecting some users. The
bulk of people seem to have everything working fine.

One extra question. I'd you apply the surround profile a second time
straight after it fails, does it still fail a second time?

Oh, and there is one more thing you could try? I read somewhere that
Guru3dcs Display Driver Uninstaller (
https://www.guru3d.com/files-details/display-driver-uninstaller-download.html)
will completely remove all traces of NVIDIA driver, and then that would let
you reinstall it as new. This would confirm that the problem wasn't caused
by a previous NVIDIA upgrade, and would ensure that you have the latest
NVIDIA settings as if you just installed a brand new video card.

I've never used DDU, so please only use DDU if you are comfortable with
fixing any issues it may cause.

Good luck, and I'll keep seeing what I can find.

Thanks
Terry

**Comment by terrymacdonald** [comment]
@FormelLMS can you please uninstall 2.2.0.52, and then try 2.1.5.37 again please? I want to know if you still get this NvAPI_Mosaic_EnableCurrentTopo(0) error changing between the regular and surround display profiles with the older 2.1.5 software. If that's the case, then I know it's nothing to do with the multithreaded singleinstance code I've added since 2.1.5.37 (which is now part of 2.2.0.52).

Should be able to use the config you have now.

Please generate a Support ZIP, and then you can upgrade to the 2.2.0.52 test version I gave you before.

Thanks
Terry

**Comment by FormelLMS** [comment]
Yes, can you send me the 2.1.5? I only find the 2.1.5.5
Sorry :(

**Comment by terrymacdonald** [comment]
@FormelLMS sorry I meant 2.1.5.5. But scratch that idea....

Instead, can you download and try the last version of DisplayMagician that you know was successfully swapping display layouts for you?  I would like to see if an old version that was working ok before still is working for you now.

The idea is to see if the NVIDIA driver has changed and stopped DisplayMagician from working for you. If the older version that was working properly in the past no longer works for you now, then we know that it is the NVIDIA driver, or something on windows that has changed to do it working.

If the older version that was working properly in the past still works for you now, then we know that it's something in the latter versions of DisplayMagician that is the cause.

Can you please give that test a go and see what you find?

Note: you don't need to delete your current Configs, just rename the c:\Users\<username>\AppData\Local\DisplayMagician folder to something like c:\Users\<username>\AppData\Local\DisplayMagicianBackup, and then install and run the version of DisplayMagician you want to test. You can then rename the folder back when you want the old Configs back again.

Thanks
Terry

**Comment by versedi** [comment]
> The idea is to see if the NVIDIA driver has changed and stopped DisplayMagician from working for you. If the older version that was working properly in the past no longer works for you now, then we know that it is the NVIDIA driver, or something on windows that has changed to do it working.

I think this is what happened to me, I had the whole PC crashing while swapping from surround to single screen. 

Today I've made even more mess by unplugging a monitor in Surround, and replugging via other cable source. I'm stucked with Display Magician being uninstalled still doing something wonky with Windows Apps and resetting my two monitors that remained unplugged. I've tried DDU - it worked until I've started AC again. 

At this point I'm going to reinstall my Windows setup cause I'm out of ideas, Even CCleaner restarts my two monitors.

**Comment by sk666** [comment]
@versedi did you tried to roll back a system restore point?

**Comment by versedi** [comment]
I didn't had one setup and from what I've tried it might have been something with broken installation of 1.x.6 that I couldn't even removed. 

Windows reinstall obviously fixed the issue, though I highly recommend everyone to avoid unplugging a monitor on one cable source while in Surround and plugging it back with other. 

It was just domino effect of bugs after that.

**Comment by terrymacdonald** [comment]
Do you think it was the unplugging of the surround that caused your
problem?

**Comment by versedi** [comment]
> Do you think it was the unplugging of the surround that caused your

@terrymacdonald #81 

As for this issue I'm going to see in upcoming days if it still requires hard restart when switching from surround to single (sometimes, it's not consistent).

**Comment by terrymacdonald** [comment]
If you could look to see if there is an error in the DisplayMagician.log
file that has "NvAPI_mosaic_enabletopo(0)" when you run the tests? If its
got the same issue then it should have the same error message.

Thanks
Terry

**Comment by FormelLMS** [comment]
Hey Terry,

so I've installed the 1.1.4 again and I can submit, that it works with this version :)

**Comment by terrymacdonald** [comment]
Ok cool! So we know that it is something in the way that the NVIDIALibrary
interacts with the Mosaic functions that isn't quite right now. It was ok
(because it was working), but now it isn't working.

The strange thing is that it is only affecting a small percentage of users.
The majority are completely fine. This makes it difficult to troubleshoot,
as we don't have enough samples to know where the problem may lie.

Do you remember exactly when you started having issues with
DisplayMagician? If you can it might give us an idea when this started
happening.

The next question is how do we troubleshoot what is broken? Well I think I
have a strategy that will work, though it will take a bit of time.

I think we should cut out as much from the configurations as possible, and
then see if the mosaic can be turned off and turned on. Then if that works
we keep adding in sections of coffee until we find the one that is causing
the issues. If I make up a sequence of versions of NVIDIAInfo with minimal
config and then more config, then we will get more of an answer out of it.

@sk666 and @FormelLMS, are you both happy to try out some more tests when
the code is ready?

Thanks
Terry

**Comment by sk666** [comment]
Hi terry,

My trouble with surround began when I started to use DM 2.x version, with your fjrst version if I remember correctly.

I will happy to help you more with that but it can take a couple of days during each test because I'm starting to be a little bit busy for a couple of week. But I will do it when I will have the time for sure :)

Thank you

Télécharger Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by terrymacdonald** [comment]
Cool, thanks for the info @sk666. Don't work too much about a quick
turnaround time. This is a tricky problem that will take a bit of time to
sort out. It will be a while before we find a solution I think.

Thanks
Terry

**Comment by FormelLMS** [comment]
Call me in to test the steps :) 
For me it start to happen with the 2.x Version. I thought it will be much work to upgrade to 2.x so I’ve waited a long time to do that. But after installing Windows 11 I‘ve updated to 2.x. 

> Am 23.01.2022 um 21:36 schrieb terrymacdonald ***@***.***>:
> 
> ﻿
> Cool, thanks for the info @sk666. Don't work too much about a quick
> turnaround time. This is a tricky problem that will take a bit of time to
> sort out. It will be a while before we find a solution I think.
> 
> Thanks
> Terry
>

**Comment by terrymacdonald** [comment]
Perfect. Thanks team.

I'll try have a think over the next few days about how I'm going to do
this. This seems like quite a tricky little problem to discover and fix.
The good thing is that you both reliably have the problem, so we should be
able to know when we've fixed it too!

Thanks
Terry

**Comment by HS34** [comment]
> Hey Terry,
> 
> so I've installed the 1.1.4 again and I can submit, that it works with this version :)

Hi guys,
Terry, congratulations for your amazing tool !

Unfortunately, in my case it doesn't work either in version 1.1.4. I've just checked it.
I'm pretty sure it worked with version 1.1.4 sometimes ago, I think before I updated my windows 10 to version 21h2 (or maybe even before 21h1).

My NVidia Driver is still 452.06 and I will stick to it as long as I can because it's the last driver compatible with 3DVision in DX11.
So my driver version has not changed for a long time. That's why I wonder if windows hasn't change something in updates.

I've got 4 screens connected on 2 TitanX (pascal) : 3x 3DVision Asus screens (for surround) + my TV (HDR enabled on the TV)
When I enable or disable surround using nvidia control panel, my PC sometimes crashes, stucked with black screens. HeliosDisplayManagement and DisplayMagician worked far better (at least for a time). Congratulations again for doing better than the manufacturer ! 

Here is what I've checked : 

-----------------------------------------------
Display Magician 1.1.4
-----------------------------------------------
created profile "3 screens only"
created profile "4 screens"
created profile "surround"
Failed to apply profile "3 screens only"
Failed to apply profile "4 screens"

I disabled surround in nvidia configuration panel and set "activate all screens" : the positions of my screens were not correct
Success applying profile "3 screens only" : the positions of my screens are now OK
Failed to apply profile "4 screens"
Failed to apply profile "surround"

-----------------------------------------------
Display Magician 2.2.0.52
-----------------------------------------------
created profile "3 screens only"
created profile "4 screens"
Failed to apply profile "3 screens only"

created profile "surround"
Failed to apply profile "3 screens only"
Failed to apply profile "4 screens"

created profile "SLI optimise performances on 1 screen"
Failed to apply profile "3 screens only"
Failed to apply profile "4 screens"
Failed to apply profile "surround"

I disabled SLI in nvidia configuration panel and set "activate all screens" : the positions of my screens were not correct
Failed to apply profile "3 screens only"
Success applying profile "4 screens" : the positions of my screens are now OK
Failed to apply profile "surround"

Regards,

**Comment by terrymacdonald** [comment]
Hi @hs34

Could you please log this as a new issue? It sounds like it's a slightly
different problem.

Please make sure that you try and make it fail, and then without closing
DisplayMagician can you please ZIP up your
C:\Users\<username>\AppData\Local\DisplayMagician\ folder, and then upload
it to the new issue that you create. That will give me some log entries to
review, and will show me what could be wrong.

Please also tell me if you have tried v2.1.5 of DisplayMagician at all. If
be very interested to see if you have the same problem with the v2.x series
of DisplayMagician as you do with the V1.1.4.

It could a absolutely be a windows change affecting your system (and us
too}, but it will be very difficult to know that. The fact that v1.1.4
works for @sk666 and @FormelLMS when the 2.x series doesn't shows me that
there is something in the way the 2.x works that doesn't work. The fact
that v1.1.4 does work for them means that it's likely a rare bug in my
NVIDIALibrary custom video library.

Thanks
Terry

**Comment by sk666** [comment]
Hi terry,

What DM version should ai use with these tests?

Thank you

Télécharger Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by terrymacdonald** [comment]
Hi @sk666,

None. Please use the NVIDIAInfo-v1.4.0-test1.zip that I've added to that
issue. We'll be testing with NVIDIAInfo for the next little while as
NVIDIAInfo was designed for testing the NVIDIALIbrary, and that's what we
need to do for the next little while. We need to avoid all the extra bits
that DisplayMagician brings, and focus on fixing the underlying
NVIDIALibrary problem, and that means using NVIDIAInfo.

Thanks

*Terry *

**Comment by terrymacdonald** [comment]
Ok Gents, some great news. I worked my way through NvAPIWrapper code based looking for things I didn't implement, or things I did wrong, and I think I found something! I have been using NvAPI_Mosaic_SetDisplayGrids, whereas NvAPIWrapper was using NvAPI_Mosaic_SetCurrentTopo. 

This doesn't seem to be much difference in practice, but I wonder if using one over the other has some longer term side effects? I've just written some new code to try and implement these missing functions, and they don't work right now. I've got some more troubleshooting to do, and then some more internal testing to make sure it works on my test systems. At the moment the NvAPI_Mosaic_SetCurrentTopo returns a TOPO_NOT_POSSIBLE error, when I'd just set it up moments before. So once I get it working I'll get you both to test it.

I'm also going to check with NVIDIA which functions are the right ones to use. Just to be sure I'm doing it the right t way.

Thanks.
Terry

**Comment by FormelLMS** [comment]
Yes, please don't forget me :) Yesterday I couldn't have a dive into it :)

**Comment by terrymacdonald** [comment]
Of course not @FormelLMS. This was a response message to @sk666.

I am definitely on a journey of discovery with this error. It looks now as
though NVIDIA recommends setting the DisplayGrid Topology to a set of
individual  1x1 display grid topologies to turn off Mosaic mode, so I am
currently trying to figure out how to do that. I'm hopeful I will have a
setup that will work within next week or so. I'll craft this into a new
NVIDIAInfo, then you'll both get to test it to make sure it works. Once
we've confirmed the display library works better, then we'll look at
updating DisplayMagician.

Thanks
Terry

**Comment by terrymacdonald** [comment]
OK @FormelLMS and @sk666, some amazingly great news! After a LOT of work I've finally cracked the problem of how to disable Mosaic/Surround without using disabling the NvAPI_Mosaic_EnableCurrentTopo. Turns out that with some clever processing I can convert the existing Mosaic grid topology to a series of single screen grid topologies using NvAPI_Mosaic_SetDisplayGrids, and the NVIDIA Driver will disable Mosaic/Surround!

I have also engineered it that if the NvAPI_Mosaic_SetDisplayGrids fails to work, it will revert to using NvAPI_Mosaic_EnableCurrentTopo instead. This now make the mosaic disabling much more robust 👍 

So, can you please both download and extract the attached NVIDIAInfo v1.5.0, and then perform the following tests for me. Make sure you do every step in order, and record the results:

[NVIDIAInfo-v1.5.0.zip](https://github.com/terrymacdonald/DisplayMagician/files/7955519/NVIDIAInfo-v1.5.0.zip)

- Open a command line/powershell to that same folder you extracted NVIDIAInfo-v1.5.0.zip
- Delete the log files from the previous tests: `del *.log`
- Delete the cfg files from the previous tests: `del *.cfg`
- Set up a new Regular profile setup, and then run `NVIDIAInfo save regular.cfg`. That will save the Regular configuration (non-surround)
- Rename the NVIDIAInfo.log file to `Rename-Item -Path "NVIDIAInfo.log" -NewName "NVIDIAInfo-regular-save.log"`
- Test that the Regular configuration was saved correctly by `NVIDIAInfo equal regular.cfg`. Please record if this test was equal or not. We want it to be equal.

That's the regular config saved. Next we want to create a new surround config.

- Set up your Surround profile setup using NVIDIA Control Panel, and then run `NVIDIAInfo save surround.cfg`. That will save the Surround configuration
- Rename the NVIDIAInfo.log file to `Rename-Item -Path "NVIDIAInfo.log" -NewName "NVIDIAInfo-surround-save.log"`
- Test that the Surround configuration was saved correctly by `NVIDIAInfo equal surround.cfg`. Please record if this test was equal or not. We want it to be equal.

At this stage we have the two basic configs set. Next step is to see if changing between the display profiles works:

- Change from the Surround profile to the Regular profile by running `NVIDIAInfo load regular.cfg`. That will try to swap to the Regular configuration.
- Rename the NVIDIAInfo.log file to `Rename-Item -Path "NVIDIAInfo.log" -NewName "NVIDIAInfo-regular-load.log"`
- Save the current profile by running `NVIDIAInfo save regular-applied.cfg`. That will save whatever the config is now, so I can compare it against what we want it to be.
- Rename the NVIDIAInfo.log file to `Rename-Item -Path "NVIDIAInfo.log" -NewName "NVIDIAInfo-regular-applied.log"`
- Test that the configuration no longer matches the Surround profile by running `NVIDIAInfo equal surround.cfg`. Please record if this test was equal or not. We want it to NOT be equal.
- Test that the Regular configuration was loaded correctly by `NVIDIAInfo equal regular.cfg`. Please record if this test was equal or not. We want it to be equal. This means the profile was matched properly.

Now, we try to swap back to Surround:

- Change from the Regular profile to the Surround profile by running `NVIDIAInfo load surround.cfg`. That will try to swap to the Surround configuration.
- Rename the NVIDIAInfo.log file to `Rename-Item -Path "NVIDIAInfo.log" -NewName "NVIDIAInfo-surround-load.log"`
- Save the current profile by running `NVIDIAInfo save surround-applied.cfg`. That will save whatever the config is now, so I can compare it against what we want it to be.
- Rename the NVIDIAInfo.log file to `Rename-Item -Path "NVIDIAInfo.log" -NewName "NVIDIAInfo-surround-applied.log"`
- Test that the configuration no longer matches the Regular profile by running `NVIDIAInfo equal regular.cfg`. Please record if this test was equal or not. We want it to NOT be equal.
- Test that the Surround configuration was loaded correctly by `NVIDIAInfo equal surround.cfg`. Please record if this test was equal or not. We want it to be equal. This means the profile was matched properly.

At this point, our tests are done. Please create a zip archive containing the following files, and upload it here to this issue:
- surround.cfg
- surround-applied.cfg
- NVIDIAInfo-surround-load.log
- NVIDIAInfo-surround-applied.log
- regular.cfg
- regular-applied.cfg
- NVIDIAInfo-regular-load.log
- NVIDIAInfo-regular-applied.log

I'll go through the results, and lets see if the new code works!

Thanks
Terry

**Comment by FormelLMS** [comment]
Terry, something strange happened... 

Would do all the steps above.
First step, the saving of the regular.cfg failed! In the zip file you can see the error message. Put the logfile in, too.

But: Display Magician is working now. I'm on 2.2.0.41 (dunno if it's latest, a bit confused now about that)
I properly could change from Normal to surround and vice versa. 

Thats very odd. But indeed I've changed something. Set some Custom DSR Resolutions into the Driver with the custom resolution tool (because I wanted to try to drive my Raceroom Racing without Surround.

**Comment by terrymacdonald** [comment]
This is indeed good news! The fact you can swap between surround and non
surround is great! I'll have a look to nail down the other parts.

This version of NVIDIAInfo also includes windows taskbar code, soon it's
possible there are still a few errors there.

Can you please also try moving your taskbar to a new location (if you're on
windows 10, I can't remember if you are), and then save a new cfg file?
Then please swap between the two different display profiles and see if the
taskbar moves places ok?

I really need to test that taskbar functionality on lots of people's
computers to be sure it works. I'd hate to fix the display profile and
introduce another bug at the same time!

Thanks
Terry

**Comment by FormelLMS** [comment]
Hey Terry

have you seen my attached zip. As I’ve tried to say, with NVIDIA info I can’t do anything because saving the normal.cfg failed. 
I’ve attached the error message in a text file in the zip. 

Switching back and forth does now work in display magician but only when I have a DSR profile enabled. 

Sorry if I have not found the right words. 

> Am 28.01.2022 um 20:15 schrieb terrymacdonald ***@***.***>:
> 
> ﻿
> This is indeed good news! The fact you can swap between surround and non
> surround is great! I'll have a look to nail down the other parts.
> 
> This version of NVIDIAInfo also includes windows taskbar code, soon it's
> possible there are still a few errors there.
> 
> Can you please also try moving your taskbar to a new location (if you're on
> windows 10, I can't remember if you are), and then save a new cfg file?
> Then please swap between the two different display profiles and see if the
> taskbar moves places ok?
> 
> I really need to test that taskbar functionality on lots of people's
> computers to be sure it works. I'd hate to fix the display profile and
> introduce another bug at the same time!
> 
> Thanks
> Terry
> 
>

**Comment by terrymacdonald** [comment]
Ah, I understand now. Looking at the error provided it's in the new TaskBar so it's a different little bug to squash. I've just squashed the bug, so can you please run the exact same tests using this NVIDIAInfo v1.5.1?

[NVIDIAInfo-v1.5.1.zip](https://github.com/terrymacdonald/DisplayMagician/files/7961968/NVIDIAInfo-v1.5.1.zip)

Thanks
Terry

**Comment by terrymacdonald** [comment]
Also @FormelLMS @sk666,

After you have both completed testing NVIDIAInfo v1.5.1 using the instructions above, can you please both try this new version of DisplayMagician I put together as well. 

This new test version of DisplayMagician v2.2.0.112 has a lot of new functionality:

- Is now a single instance, so games will start faster and display profiles will swap quicker
- Is now multithreaded to handle processing profile changes in the background
- Display layout detection is more reliable
- Now uses a new way of turning off NVIDIA surround so it avoids crashes on some video cards
- Stores the windows taskbar location and settings and attempts to reposition it to the correct place to fix NVIDIA driver issues
- Now allows you to uninstall the DisplayMagician Desktop Background Context Menu if you don't like it.

Please test it out, and tell me if there are any bugs. NOTE: There are quite a few changes in v2.2.0.112 compared to v2.1.5.5, so you will be asked to recreate new display profiles again. Please make sure you do this so that everything will work ok.
[DisplayMagicianSetup-v2.2.0.112.zip](https://github.com/terrymacdonald/DisplayMagician/files/7962890/DisplayMagicianSetup-v2.2.0.112.zip)

Thanks
Terry

**Comment by FormelLMS** [comment]
So we now are steps ahead :) 

Tests with NvidiaInfo 1.5.1:
After Applying the regular.cfg I got those error Message:


Attempting to apply NVIDIA display config from regular.cfg
ERROR - NVIDIAInfo Windows CCD settings were not applied correctly.

Equality checks then of surround and regular.cfg are NOT equal after that
Taskbar was on the left monitor instead of the middle

After changing back to surround profile, the surround.cfg was equal
[NvidiaInfo 1.5.1 Test.zip](https://github.com/terrymacdonald/DisplayMagician/files/7963530/NvidiaInfo.1.5.1.Test.zip)


Tests with the new Display Magician:
Now it seems fine withour Custom DSR Resolutions. I can delete them and its working to switch between surround and not surround.

What is nor Working:
I have now three Configs
Surround 1440p
Normal 1440p
Normal 1080p (I only have 1080p Monitors)

Changing from Surround an Normal 1440p and vice versa is working
Changing from Surround to Normal 1080p does switch to 1440p resolution, but surround off
Changing from Normal 1440p to Normal 1080p doesn't switch the text resolution (for 1440p it's on 125 percent and for 1080p it's 100 percent)

**Comment by terrymacdonald** [comment]
Hi @FormelLMS

I don't think DisplayMagician stores the text sizing at all. I'd have to
double-check to confirm, but I think that may be in the GetDisplayConfig
function, and we don't currently get that information.

I've tried making a C# function to access GetDisplayConfig and
SetDisplayConfig in the past, but I could never get it to work. I'm a lot
more comfortable writing Process Invoke fictions now, so I might revisit
that.

I guess what I'm saying is I may not be able to get the text size working.
I'll take a look at it, but I'll have to judge how tricky it will be and
how much it will affect the rest of the DisplayMagician codebase.

Thanks
Terry

**Comment by terrymacdonald** [comment]
@FormelLMS, can you please send me a DisplayMagician support zip, and a
screenshot showing where you set the text sizing in the NVIDIA Control
panel? This will help me find where the information is in the NVAPI.

Thanks
Terry

**Comment by terrymacdonald** [comment]
Gah! I'll have to hold off on the Get/SetDisplayConfig work as I've found a bug in the way that Windows handles taskbar position changes on daisy-chained DisplayPort screens. Windows Settings doesn't update registry for the 2nd connected display (the one daisy-chained from the first one), meaning that I can't just record the taskbar settings as they could be wrong :(. So I now have to do some changes to store the actual _intent_ of the user, and then make my code brute force those details into reality. 

Windows does get frustrating at times :(.

**Comment by sk666** [comment]
Hi Terry,

sorry for the delay. I've got an error in powershell but thanks to @FormelLMS the error was because I did not used the  .\ before the command line.

I've got my result from nvidiainfo 1.51, see atached files. I will try later today with DM 2.2.0.112

Thank you
[Results sk666.zip](https://github.com/terrymacdonald/DisplayMagician/files/7965988/Results.sk666.zip)

**Comment by terrymacdonald** [comment]
Thanks sk666.

**Comment by sk666** [comment]
hi Terry,

v2.2.0.112 od DM is totally unusable on my computer, it faild at the opening and DM can not bypass this error so I can not create a log zip file. I need to kill it by the task manager after the error spawn.

The error is in french but I think in english is something like: "The value can not be null. Paramter name: first".

Did you ever seen that before?

Thank you


![fatal error at openning](https://user-images.githubusercontent.com/29050341/151726073-d09d8cfa-37fe-4078-bbe5-81194ea4e5a7.PNG)

**Comment by terrymacdonald** [comment]
Hi @sk666,

Can you please delete the existing DisplayProfile_2.2.json file you having
in your DisplayMagician app folder? The updating process assumes you don't
have a DisplayProfile_2.2.json file, but as you've been helping me you
already have that file

If you delete the existing DisplayProfile_2.2.json file from your
DisplayMagician appdata folder then it should start up properly.

Thanks
Terry

**Comment by FormelLMS** [comment]
Hey Terry, here you can find the support zip.
Text Scaling isn't in nvidia driver, it's in the display options.
[DisplayMagician-Support-20220131-2019.zip](https://github.com/terrymacdonald/DisplayMagician/files/7973563/DisplayMagician-Support-20220131-2019.zip)
![Screenshot 2022-01-31 201956](https://user-images.githubusercontent.com/87239140/151858535-ab662f64-e09e-487b-b925-3006bcc7c198.jpg)

**Comment by sk666** [comment]
Hi Terry,

I found the 2.2.json file, I deleted it, and started agian DM. It recreate it and then crash with the same error :(

Thank you

Télécharger Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by terrymacdonald** [comment]
Oh shoot. That's not good. I think it's indicating that there is a config
element that is null that is not supposed to be null. Can you please upload
the DisplayMagician.log file, and any other DisplayMagician1.log -
DisplayMagician4.log log files you find in the
C:\Users\<username>\AppData\Local\DisplayMagician\Logs folder please? You
may need to zip them up. This will let me see what the error is.

Thanks

*Terry MacDonald *| Principal Security Consultant | Chief Operating Officer



M: +64 211 918 814 <+64+211+918+814>
E: ***@***.***
W: www.cosive.com

**Comment by sk666** [comment]
@terrymacdonald there is the log files.

thank you


[logs.zip](https://github.com/terrymacdonald/DisplayMagician/files/7982267/logs.zip)

**Comment by terrymacdonald** [comment]
Weird. This is complaining about missing colour information from your screens in your configuration. That shouldn't happen. 

```
2022-01-31 19:53:43.0618|ERROR|DisplayMagician.Program|Program/StartUpApplication top level exception: La valeur ne peut pas 괲e null.
Nom du param贲e : first:    ࠓystem.Linq.Enumerable.SequenceEqual[TSource](IEnumerable`1 first, IEnumerable`1 second, IEqualityComparer`1 comparer)
   ࠄisplayMagicianShared.NVIDIA.NVIDIA_COLOR_CONFIG.Equals(NVIDIA_COLOR_CONFIG other)
   ࠄisplayMagicianShared.NVIDIA.NVIDIA_DISPLAY_CONFIG.Equals(NVIDIA_DISPLAY_CONFIG other)
   ࠄisplayMagicianShared.ProfileItem.Equals(ProfileItem other)
   ࠄisplayMagicianShared.ProfileRepository.UpdateActiveProfile()
   ࠄisplayMagicianShared.ProfileRepository.get_CurrentProfile()
   ࠄisplayMagicianShared.ProfileItem.get_IsActive()
   ࠄisplayMagician.UIForms.MainForm.RefreshNotifyIconMenus()
   ࠄisplayMagician.UIForms.MainForm..ctor(Form formToOpen)
   ࠄisplayMagician.Program.StartUpApplication() - 
```

The problem is with the Dell 2208WFP screen with UID4356 connected via DVI:

```
WINAPI|\\?\PCI#VEN_10DE&DEV_13C2&SUBSYS_85081043&REV_A1#4&80e7924&0&0008#{5b45201d-f2f2-4f3b-85bb-30ff1f953599}|DISPLAYCONFIG_OUTPUT_TECHNOLOGY_DVI|44048|16444|\\?\DISPLAY#DEL403C#5&28d4f9e&1&UID4356#{e6f07b5f-ee97-4a90-b076-33f57bf4eaa7}|DELL 2208WFP
```

I believe that this is because DVI doesn't support accessing the colour information through it. I think this may be the source of part of your past problems, and definitely a bug with NVIDIALibrary, AMDLibrary and WinLibrary! This is a great find!

I will see what I can do to not grab colour settings with some of the older style screens. This might help avoid some of the issues with older hardware until the newer hardware becomes the norm.

Thanks
Terry

**Comment by sk666** [comment]
Hi Terry,

don't loose your precious time on that, I can live with the 1.1.4 version od DM. I will change my 2 old monitor eventually, so I will not use DVi with my new screens. If it's possible to stay with 1.1.4 with nvidia and w10, it will do the job for me. My computer could not have w11 so I will need to change it too in the next 3 years.

Thank you

**Comment by terrymacdonald** [comment]
No no, it's something fairly easy to add, and I want everyone to be able to
use the new faster DM. It's just a little more work, and it will mean I
avoid so many other future issues.

Thanks
Terry

**Comment by terrymacdonald** [comment]
@sk666  and @FormelLMS  - some more tests to perform please! I've just finished getting taskbar settings to record and I'm pretty happy with how they turned out :). I've also configured DisplayMagician to ignore getting HDR/Colour settings from the following types of connections: VGA, COMPONENT, SVIDEO, DVI, COMPOSITE. 

@sk666 that should stop you getting errors (hopefully), so it would be cool if you could both install DisplayMagician v2.2.0.179 (yes this is build number 179!) that I've attached here.

IMPORTANT: YOU WILL NEED TO DELETE YOUR DisplayProfiles_2.2.json file AND CREATE NEW PROFILES FROM SCRATCH.

This version includes changed DisplayProfile format again in order to track all the new features we've been working on recently. It will require new display profiles.

[DisplayMagicianSetup-v2.2.0.179.zip](https://github.com/terrymacdonald/DisplayMagician/files/8001419/DisplayMagicianSetup-v2.2.0.179.zip)

Please test this out and see if there are any broken bits. If there are, please upload a Support ZIP File with a description of what's wrong. 

I think we're getting really close to a new release!

Thanks
Terry

**Comment by sk666** [comment]
Hi Terry,

wow build 179. It was a lot of work, but I'm glad to say that this build fix the switching profile bug, and it resolve my bog with DVI when I start DM :) I hope that it will fix the issue too for @FormelLMS 

Issue 75 still persist for me, even if the app is minimized to system tray or not. But at least, DM works again, thank you very much :)

**Comment by terrymacdonald** [comment]
@sk666 @FormelLMS  I fixed yet more issues, so now please test DisplayMagician v2.2.0.223 (attached). This new version shows the taskbars in the DM UI, and also lets you easily change your taskbar layout using the new 'Profile Tools' button on the Display Profiles page :). 

![image](https://user-images.githubusercontent.com/5721772/152664225-ebb30318-e255-40d4-8460-1c968acaa59b.png)

Can I please trouble you to do some testing for me again? I *think* I have fixed the issues you mentioned , but I really need confirmation from you that it is now working.

IMPORTANT: Just to be sure, can you delete all your C:\Users<username>\AppData\Local\DisplayMagician\Profiles\DisplayProfiles_*.json files just so we know that all the configs are generated in the right config format? This will let me make sure it is working in a known state and avoid some of the past bugs.

Then please download DisplayMagician v2.2.0.223 and install it. Then set up new Display Profiles, and then use those profiles in the Game Shortcuts. 
[DisplayMagicianSetup-v2.2.0.223.zip](https://github.com/terrymacdonald/DisplayMagician/files/8009215/DisplayMagicianSetup-v2.2.0.223.zip)
 
Please test the following:
- Test changing between surround and non-surround profiles
- Please test changing taskbar position (can do that using the new 'Profile Tools' button under Profiles.
- Please check the UI shows the correct display layout and that the taskbar layout in the UI matches the real thing.

And then please report back with what you've found.

Thanks
Terry

**Comment by sk666** [comment]
Hi Terry,

everything seems to be working fine for me with build .223 including taskbar..except for what I said previously in my last reply on issue #75.

thank you

**Comment by terrymacdonald** [comment]
Hi @sk666,

Can you please create a new Support Zip file in 223 and upload it under
issue  #75? It's like to double check to see what the final issue is now
we've solved the others.

Thanks
Terry

**Comment by FormelLMS** [comment]
Hey Terry, I'm sorry, I was away a few days. Tried now the new version and everything is working well. This is fantastic. 
Changed taskbar to the top in normal mode. In Surround mode it's not working, but I think this is normal. Taskbar is a bit strange in behaviour of surround :)
Changing the profiles is very fast now.

Thank you very much!

**Comment by terrymacdonald** [comment]
Ok, I'm going to close this issue as it is fixed in the codebase. I'll concentrate on getting the taskbar location sorted.
